src/libraries/Microsoft.PowerFx.Interpreter/PatchFunction.cs

```diff
@@ -0,0 +1,204 @@
+        }
+
+        public override RequiredDataSourcePermissions FunctionPermission => RequiredDataSourcePermissions.Create | RequiredDataSourcePermissions.Update;
+    }
+
+    internal class PatchFunction : PatchAndValidateRecordFunctionBase, IAsyncTexlFunction
+    {
+        public override bool IsSelfContained => false;
+
+        public PatchFunction()
+            : base("Patch", TexlStrings.AboutPatch, FunctionCategories.Table | FunctionCategories.Behavior, DType.EmptyRecord, 0, 3, int.MaxValue, DType.EmptyTable, DType.EmptyRecord, DType.EmptyRecord)
+        {
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.AboutPatch, TexlStrings.AboutPatch, TexlStrings.AboutPatch };
+            yield return new[] { TexlStrings.AboutPatch, TexlStrings.AboutPatch, TexlStrings.AboutPatch, TexlStrings.AboutPatch };
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures(int arity)
+        {
+            if (arity > 3)
+            {
+                return GetGenericSignatures(arity, TexlStrings.AboutPatch, TexlStrings.AboutPatch, TexlStrings.AboutPatch);
+            }
+
+            return base.GetSignatures(arity);
+        }
+
+        public async Task<FormulaValue> InvokeAsync(FormulaValue[] args, CancellationToken cancel)
+        {
+            var argFields = new Dictionary<string, FormulaValue>();
+            var fieldsDict = new Dictionary<string, FormulaValue>();
+
+            var arg0 = args[0] as TableValue;
+            var arg1 = args[1] as RecordValue;
+
+            // Starts from the 3rd argument (new values)
+            for (var i = 2; i < args.Length; i++)
+            {
+                var arg = args[i];
+
+                if (arg is BlankValue)
+                {
+                    continue;
+                }
+                else if (arg is ErrorValue)
+                {
+                    // @@@JYL How to handle error value arguments?
+                    return arg;
+                }
+                else if (arg is RecordValue record)
+                {
+                    foreach (var field in record.Fields)
+                    {
+                        argFields[field.Name] = field.Value;
+                    }
+                }
+            }
+
+            var fieldList = new List<NamedValue>();
+
+            // Creates final record 
+            foreach (var field in arg1.Fields)
+            {
+                var namedValue = argFields.ContainsKey(field.Name) ? argFields[field.Name] : field.Value;
+
+                fieldList.Add(new NamedValue(field.Name, namedValue));
+            }
+
+            var newRecord = FormulaValue.NewRecordFromFields(fieldList);
+
+            // @@@JYL Whats a good way to replace the old record with the new one and "update" the TableValue object?
+
+            var deleted = await arg0.RemoveAsync(arg1);
```

Comment:
This should probably be a "UpdateField" virtual on RecordValue. 

Not a remove+append.

Reply:
another reason is that Remove/append will change the order of the collection.